camus-etl-kafka/src/main/java/com/linkedin/camus/etl/kafka/mapred/EtlRecordReader.java

```diff
@@ -65,13 +66,16 @@
    * @throws IOException
    * @throws InterruptedException
    */
-  public EtlRecordReader(InputSplit split, TaskAttemptContext context) throws IOException, InterruptedException {
+  public EtlRecordReader(EtlInputFormat inputFormat, InputSplit split, TaskAttemptContext context) 
```

Comment:
Same thing: I think inputFormat can be replaced by Class<? extends InputFormat> inputFormatClass. This way, you don't need line 36 and 71. Also it's safer since there's no chance to accidentally modify inputFormat.